tempel.el

```diff
@@ -580,7 +581,17 @@ This is meant to be a source in `tempel-template-sources'."
   "Template completion is done."
   (interactive)
   ;; TODO disable only the topmost template?
-  (while tempel--active (tempel--disable)))
+  (while tempel--active (tempel--done)))
+
+(defun tempel--done (&optional st)
+  "Finalize template ST, or last template."
+  (let* ((st (or st (car tempel--active)))
+         (range (caar st))
+         (env (cdr st))
+         (buffer (current-buffer)))
+    (eval (overlay-get range 'tempel--post) env)
+    (with-current-buffer buffer
```

Comment:
Fwiw, the `(with-current-buffer ...)` is required here because the `:post` callback may change the buffer and that would confuse the subsequent `(tempel--disable st)` below. The `elisp` template with the call to `(org-edit-src-code)` in the README was a great test case for this 😅

Reply:
Yes, I know. (I was about to propose save-excursion.)

Reply:
That's even better, I agree; updated.

Reply:
Oh sorry! You misunderstood what I meant. I think as you had it was better since the post action may want to move the point.

Reply:
Oh, I see now. I rolled back the previous change from `with-current-buffer` to `save-excursion`. Good thinking re. the post action moving point! 👍